### PR TITLE
Add index for queries in the JSON output to fix the order of the queries

### PR DESCRIPTION
**Pull request by blishko**
This PR represents the first step towards the goal of understanding when a change in the encoding happens and manifests in the JSON output tests.
Here we try to fix the order of the unanswered queries in the JSON object by adding a prefix counter before the hash.
When the answers are provided, the prefix is removed (if still present) upon reading the JSON object.

### CODE REVIEW

File: libsolidity/interface/StandardCompiler.cpp [hunk 1]
```diff
@@ -613,6 +615,8 @@ std::variant<OptimiserSettings, Json::Value> parseOptimizerSettings(Json::Value
 
 }
 
+char const queryKeyDelimiter = '#';
```
Comment (leonardoalt) [inline libsolidity/interface/StandardCompiler.cpp:618]: Hmm this feels a bit off place. I think we can inline where it's used

File: libsolidity/interface/StandardCompiler.cpp [hunk 2]
```diff
@@ -719,8 +723,11 @@ std::variant<StandardCompiler::InputsAndSettings, Json::Value> StandardCompiler:
 			if (!smtlib2Responses.isObject())
 				return formatFatalError("JSONError", "\"auxiliaryInput.smtlib2responses\" must be an object.");
 
-			for (auto const& hashString: smtlib2Responses.getMemberNames())
+			for (auto const& key: smtlib2Responses.getMemberNames())
 			{
+				auto delimiterPos = key.find(queryKeyDelimiter);
+				auto hashStart = delimiterPos == string::npos ? 0 : delimiterPos + 1;
```
Comment (leonardoalt) [inline libsolidity/interface/StandardCompiler.cpp:729]: Shouldn't we rather assert that the delimiter is there?

File: libsolidity/interface/StandardCompiler.cpp (hunk 1 above)
Comment (blishko) [reply libsolidity/interface/StandardCompiler.cpp:618]: You mean just use '#' character on both places where this is needed and do not introduce the constant?

File: libsolidity/interface/StandardCompiler.cpp (hunk 2 above)
Comment (blishko) [reply libsolidity/interface/StandardCompiler.cpp:729]: That's possible, I thought it's better to be more benevolent. At this point the prefix is not needed. The only place where it is needed is to fix order in the JSON object.

File: libsolidity/interface/StandardCompiler.cpp (hunk 1 above)
Comment (leonardoalt) [reply libsolidity/interface/StandardCompiler.cpp:618]: Yea

File: libsolidity/interface/StandardCompiler.cpp (hunk 2 above)
Comment (leonardoalt) [reply libsolidity/interface/StandardCompiler.cpp:729]: Ok, that's also fine

**Comment by axic** [comment]
Shouldn't this be a design discussion? Wouldn't the ideal solution be making the queries an array?

**Comment by leonardoalt** [comment]
@axic actually I wanted to bring this up in the design discussion to be removed entirely

**Comment by leonardoalt** [comment]
We could go ahead with this PR by making it an array. It's a minimal change and allows us to fix the order for now.

**Comment by blishko** [comment]
OK, I also like the array solution better.
It's a change in the JSON structure, but at least we may see if anybody is using this feature :)

**Comment by blishko** [comment]
Does it make sense to implement the change to the array?
If we just remove the queries from the JSON output (if a solver is present) and thus remove the check from command line tests, then it is not necessary to change anything in the JSON structure, no?

If we *would* implement the change, then "smtlib2queries" would become an array of objects with two fields, "hash" and "query"? Is that the intended design?

**Comment by leonardoalt** [comment]
Yea, let's go with removing the queries if a solver was present.

**Comment by cameel** [comment]
What's the status here? Does it need more reviews or is it waiting for fixes?

**Comment by blishko** [comment]
Has any decision about outputting the queries been made?
I could finish the change to store the queries as array instead of object, but I am not sure if it is needed anymore.

**Comment by leonardoalt** [comment]
I think this isn't needed anymore, so I'll close it for now. We can restart the discussion at some point if needed.

**Pull request closed without merging**